src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -1096,6 +1096,20 @@ void Thread::SetActivationPending(bool isPending)
     }
 }
 
+#ifdef FEATURE_EVENT_TRACE
+const GUID* Thread::GetActivityId() const
```

Comment:
Can this be a thread_local variable in the eventpipe? I do not see a reason why this needs to be stored on the core runtime thread instance.

Reply:
One reason to have this on Thread is if eventpipe envisions a scenario where it needs to handle one OS thread attached to multiple runtimes (and thus having different Thread instances and different ActivityIDs in different runtimes). 

If ActivityID is a process-wide concept, then it should live as a threadstatic on the eventpipe side.

Reply:
I do not think EventPipe is designed for this kind of 1:N relationship with runtime. It has number of assumptions about being 1:1 with runtime.

Reply:
Can we just define `EP_THREAD_INCLUDE_ACTIVITY_ID` and use the functionality in shared eventpipe code? That way we can just use the existing `thread_local EventPipeThreadHolder` to get at the `EventPipeThread` and its `activity_id` field.

Reply:
Switched to use the EventPipe one, similar to Mono. Not sure on the ETW side though.

Reply:
Created #89657 to track the `ETW ``activity_id`